flows/engine/assets.go

```diff
@@ -24,7 +24,7 @@ type sessionAssets struct {
 var _ flows.SessionAssets = (*sessionAssets)(nil)
 
 // NewSessionAssets creates a new session assets instance with the provided base URLs
-func NewSessionAssets(source assets.Source) (flows.SessionAssets, error) {
+func NewSessionAssets(source assets.Source, migrationConfig *definition.MigrationConfig) (flows.SessionAssets, error) {
```

Comment:
Just a thought, but should the migration config just be part of the asset source? Is there any scenario where the config is different given the same source? Just wondering why those would be separate, since the source has the old definition flows it seems appropriate it would also know the config needed to do the migration?

Don't feel strongly but interface just feels slightly odd, but admit I may not have the full perspective there.

Reply:
The way things are setup currently, asset sources don't know anything about the flow versions - at an asset level a flow is just a JSON blob with a UUID and name.. it's the `FlowAssets` object in `SessionAssets` that reads and caches flow definitions. So `AssetSource` could own the migration config.. but it would never use it.. so I guess it feels more natural as an argument to `SessionAssets`

Reply:
AssetSources don't really use any of their things though, it's just an interface to hand stuff over isn't it? I don't know, to me it feels as much a part of data as anything else. Again if there is no need to ever have the same source using a different migration config then why separate.

Anyways, approval given! :)